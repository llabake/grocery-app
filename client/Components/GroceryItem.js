import React, { Component } from 'react';
import { connect } from "react-redux";
import {
  deleteGroceryItem,
  updateGroceryItem,
} from "../actions/groceryItemAction";

export class GroceryItem extends Component {
  delete = (event) => {
    event.preventDefault();
    this.props.deleteGroceryItem(this.props.item._id);
  };

  togglePurchased = (event) => {
    event.preventDefault();
    this.props.updateGroceryItem(this.props.item)
  };

  render() {
    return (
      <div>
        <div>
          <h4 className={this.props.item.purchased ? 'strikethrough' : ''}>
            {this.props.item.name}
          </h4>
        </div>
        <form onSubmit={this.togglePurchased}>
          <button type="submit" className={this.props.item.purchased ? '' : 'button-primary'}>
            {this.props.item.purchased ? 'Unbuy' : 'Buy'}
          </button>
        </form>
        <form className='' onSubmit={this.delete}>
          <button type="submit">&times;</button>
        </form>
      </div>
    );
  }
}
export const mapDispatchToProps = dispatch => ({
  deleteGroceryItem: item => dispatch(deleteGroceryItem(item)),
  updateGroceryItem: item => dispatch(updateGroceryItem(item)),
});

export default connect(null, mapDispatchToProps)(GroceryItem)
