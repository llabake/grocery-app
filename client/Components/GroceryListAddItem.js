import React, { Component } from 'react';
import { connect } from "react-redux";
import { addGrocery } from "../actions/groceryItemAction";


export class GroceryListAddItem extends Component {
  state = {
    item: {
      name: '',
    },
  };

  handleInputChange = (event) => {
    this.setState({
      item: { name: event.target.value },
    })
  };

  handleAddItem = (event) => {
    event.preventDefault();
    this.props.addGrocery(this.state.item)
      .then(() => this.setState(
        { item: { name: '' } },
      ))
  };

  render() {
    return (
      <div className='grocery-addItem'>
        <form onSubmit={this.handleAddItem}>
          <input
            value={this.state.item.name}
            onChange={this.handleInputChange}
          />
          <button type="submit">Add Item</button>
        </form>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  addGrocery: grocery => dispatch(addGrocery(grocery)),
});

export default connect(null, mapDispatchToProps)(GroceryListAddItem)