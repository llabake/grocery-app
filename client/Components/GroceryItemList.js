import React from 'react';

import GroceryListAddItem from "./GroceryListAddItem";
import GroceryItem from "./GroceryItem";

const GroceryItemList = ({ items }) => (
  <div>
    <h1 className='title'>Grocery List</h1>
    <div>
      {
        items.map(item => (
          <GroceryItem
            item={item}
            key={`item + ${item._id}`}
          />
        ))
      }
    </div>
    <GroceryListAddItem />
  </div>
);

export default GroceryItemList;