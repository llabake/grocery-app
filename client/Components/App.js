import React, { Component } from 'react';
import { connect } from "react-redux";

import GroceryItemList from "./GroceryItemList";
import '../styles.css';


export class App extends Component {
  render() {
    return (
      <div>
        <GroceryItemList items={this.props.items} />
      </div>
    )
  }
}

export default connect(state => ({ items: state.groceryItemReducer }))(App);