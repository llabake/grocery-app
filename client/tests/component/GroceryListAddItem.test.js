import React from 'react';
import { shallow } from 'enzyme'
import { GroceryListAddItem, mapDispatchToProps } from "../../Components/GroceryListAddItem";

describe('<GroceryListAddItem />', () => {
  const dispatch = jest.fn();
  const props = {
    addGrocery: jest.fn(() => new Promise(resolve => resolve())),
  };

  test('it renders correctly', () => {
    const wrapper = shallow(<GroceryListAddItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('handle change', () => {
    const wrapper = shallow(<GroceryListAddItem {...props} />);
    const spy = jest.spyOn(wrapper.instance(), 'handleInputChange');
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: 'firstName',
      },
    };
    wrapper.instance().handleInputChange(event);
    expect(spy).toHaveBeenCalled();
  });
  test('handle add item', () => {
    const wrapper = shallow(<GroceryListAddItem {...props} />);
    const spy = jest.spyOn(wrapper.instance(), 'handleAddItem');
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: 'firstName',
      },
    };
    wrapper.instance().handleAddItem(event);
    expect(spy).toHaveBeenCalled();
  });
  test('should return required actions', () => {
    const actions = mapDispatchToProps(dispatch);
    actions.addGrocery();
    expect(dispatch).toHaveBeenCalledTimes(1);
  })
});