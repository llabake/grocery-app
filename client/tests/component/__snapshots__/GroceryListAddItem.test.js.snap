// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroceryListAddItem /> it renders correctly 1`] = `
<div
  className="grocery-addItem"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      value=""
    />
    <button
      type="submit"
    >
      Add Item
    </button>
  </form>
</div>
`;
